zivid_camera/test/test_zivid_camera.cpp

```diff
@@ -657,6 +679,61 @@ TEST_F(CaptureOutputTest, testCapture2D)
   verify_image_and_camera_info(*color_image_color_sub.lastMessage(), *color_camera_info_sub.lastMessage());
 }
 
+class CaptureAndSaveTest : public TestWithFileCamera
+{
+protected:
+  void capture_and_save_to_path(const std::string& file_path, bool expected_result)
+  {
+    AllCaptureTopicsSubscriber allCaptureTopicsSubscriber(*this);
+
+    enableFirst3DAcquisition();
+    zivid_camera::CaptureAndSave capture_and_save;
+    capture_and_save.request.file_path = file_path;
+    if (expected_result)
+    {
+      ASSERT_TRUE(ros::service::call(capture_service_and_save_name, capture_and_save));
+      ASSERT_TRUE(boost::filesystem::exists(file_path));
+    }
+    else
+    {
+      ASSERT_FALSE(ros::service::call(capture_service_and_save_name, capture_and_save));
+      ASSERT_FALSE(boost::filesystem::exists(file_path));
+    }
+    short_wait_duration.sleep();
+    allCaptureTopicsSubscriber.assert_num_topics_received(1);
```

Comment:
I would throw `allCaptureTopicsSubscriber.assert_num_topics_received(0);` before the service is invoked above. Then we know that the topics were sent as a result of the capture you invoked.

Reply:
done